extension/api.js

```diff
@@ -64,67 +35,105 @@
 		return api;
 	})();
 
-	window.gitHubNotifCount = cb => {
+	window.GitHubNotify.requestPermission = permission => {
+		return new Promise(resolve => {
+			chrome.permissions.request({
+				permissions: [permission]
+			}, granted => {
+				window.GitHubNotify.settings.set(`${permission}_permission`, granted);
+				resolve(granted);
+			});
+		});
+	};
+
+	window.GitHubNotify.queryPermission = permission => {
+		return new Promise(resolve => {
+			chrome.permissions.contains({
+				permissions: [permission]
+			}, resolve);
+		});
+	};
+
+	window.GitHubNotify.request = (url, customHeaders) => {
 		const token = window.GitHubNotify.settings.get('oauthToken');
-		const opts = {
+		if (!token) {
+			return Promise.reject(new Error('missing token'));
+		}
+
+		const headers = Object.assign({
 			Authorization: `token ${token}`
-		};
-		const query = [];
-		if (window.GitHubNotify.settings.get('useParticipatingCount')) {
-			query.push('participating=true');
+		}, customHeaders);
+
+		return fetch(url, {headers});
+	};
+
+	window.GitHubNotify.getApiUrl = () => {
+		const rootUrl = window.GitHubNotify.settings.get('rootUrl');
+
+		if (/(^(https:\/\/)?(api\.)?github\.com)/.test(rootUrl)) {
+			return 'https://api.github.com/notifications';
 		}
-		query.push('per_page=1');
+		return `${rootUrl}api/v3/notifications`;
+	};
 
-		let url = window.GitHubNotify.settings.get('rootUrl');
+	window.GitHubNotify.getTabUrl = () => {
+		let rootUrl = window.GitHubNotify.settings.get('rootUrl');
 
-		if (!token) {
-			cb(new Error('missing token'), null, window.GitHubNotify.defaults.interval);
-			return;
+		if (/api.github.com\/$/.test(rootUrl)) {
```

Comment:
Nevermind. Scrolled down and see that it was moved from below. This should work as-is.